cmd/clusterctl/cmd/init.go

```diff
@@ -0,0 +1,96 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+type initOptions struct {
+	kubeconfig              string
+	coreProvider            string
+	bootstrapProviders      []string
+	infrastructureProviders []string
+	targetNamespace         string
+	watchingNamespace       string
+	force                   bool
+}
+
+var io = &initOptions{}
+
+var initCmd = &cobra.Command{
+	Use:   "init",
+	Short: "Initialize a management cluster for Cluster API",
+	Long: LongDesc(`
+		Initialize a management cluster for Cluster API by installing cluster API core components,
+		the kubeadm bootstrap provider, and the selected bootstrap and infrastructure providers.
+
+		The management cluster should be an existing Kubernetes cluster, and the identity used
+		for accessing the cluster must have enough privileges for installing cluster API providers.
+
+		Use 'clusterctl config providers' to get the list of available providers; if necessary, edit
+		the $HOME/.clusterctl.yaml file to add new provider configurations or to customize existing ones.
+		
+		Some providers require environment variable to be set before running clusterctl init; please
+		refer to the provider documentation or use 'clusterctl config provider [name]' to get the
+		list of required variables.
+
+		See https://cluster-api.sigs.k8s.io/ for more details.`),
+
+	Example: Examples(`
+		# Initialize a management cluster, the cluster currently pointed by kubectl, by installing the
+		# AWS infrastructure provider. Please note that this command, when executed on an empty management cluster, 
+ 		# automatically triggers the installation of the cluster-api core provider.
+		clusterctl init --infrastructure=aws
+
+		# Initialize a management cluster with a specific version of the AWS infrastructure provider.
+		clusterctl init --infrastructure=aws:v0.4.1
+
+		# Initialize a management cluster, the cluster defined in the "foo.yaml" kubeconfig file, by 
+		# installing the AWS infrastructure provider.
+		clusterctl init --kubeconfig=foo.yaml  --infrastructure=aws
+
+		# Initialize a management cluster, by installing both the AWS and the vSphere infrastructure provider
+		clusterctl init --infrastructure=aws,vsphere
+
+		# Initialize a management cluster by installing the provider's components' in the "foo" namespace.
+		clusterctl init --infrastructure aws --target-namespace foo
+
+		# Initialize a management cluster and configures all the providers for watching Cluster API
+		# objects in the "foo" namespace only.
+		clusterctl init --infrastructure aws --watching-namespace=foo`),
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runInit()
+	},
+}
+
+func init() {
+	initCmd.Flags().StringVarP(&io.kubeconfig, "kubeconfig", "", "", "Path to the kubeconfig file to use for accessing the management cluster. If empty, default rules for kubeconfig discovery will be used")
```

Comment:
It would be nice to have [optional] or [required] in the help text so I know what I can get away with not defining. Not a blocker, can definitely be improved later.

Reply:
I see your point and this can be a nice addition in a follow-up PR